test/config_parser_SUITE.erl

```diff
@@ -1195,7 +1244,1557 @@ s2s_max_retry_delay(_Config) ->
+    M = fun(K, V) -> modopts(mod_adhoc, [{K, V}]) end,
+    T = fun(K, V) -> #{<<"modules">> => #{<<"mod_adhoc">> => #{K => V}}} end,
+    %% report_commands_node is boolean
+    [?_eqf(M(report_commands_node, true), T(<<"report_commands_node">>, true)),
+     ?_eqf(M(report_commands_node, false), T(<<"report_commands_node">>, false)),
+     %% not boolean
+     ?_errf(T(<<"report_commands_node">>, <<"hello">>))].
+
+mod_auth_token(_Config) ->
+    check_iqdisc(mod_auth_token),
+    run_multi(mod_auth_token_cases()).
+
+mod_auth_token_cases() ->
+    P = fun(X) ->
+                     Opts = #{<<"validity_period">> => X},
+                     #{<<"modules">> => #{<<"mod_auth_token">> => Opts}}
+             end,
+    [?_eqf(modopts(mod_auth_token, [{{validity_period,access},  {13,minutes}},
+                                    {{validity_period,refresh}, {31,days}}]),
+           P([#{<<"token">> => <<"access">>,  <<"value">> => 13, <<"unit">> => <<"minutes">>},
+              #{<<"token">> => <<"refresh">>, <<"value">> => 31, <<"unit">> => <<"days">>}])),
+     ?_errf(P([#{<<"token">> => <<"access">>,  <<"value">> => <<"13">>, <<"unit">> => <<"minutes">>}])),
+     ?_errf(P([#{<<"token">> => <<"access">>,  <<"value">> => 13, <<"unit">> => <<"minute">>}])),
+     ?_errf(P([#{<<"token">> => <<"Access">>,  <<"value">> => 13, <<"unit">> => <<"minutes">>}])),
+     ?_errf(P([#{<<"value">> => 13, <<"unit">> => <<"minutes">>}])),
+     ?_errf(P([#{<<"token">> => <<"access">>,  <<"unit">> => <<"minutes">>}])),
+     ?_errf(P([#{<<"token">> => <<"access">>,  <<"value">> => 13}]))].
+
+
+mod_bosh(_Config) ->
+    run_multi(mod_bosh_cases()).
+
+mod_bosh_cases() ->
+    T = fun(K, V) -> #{<<"modules">> => #{<<"mod_bosh">> => #{K => V}}} end,
+    M = fun(K, V) -> modopts(mod_bosh, [{K, V}]) end,
+    [?_eqf(M(inactivity, 10), T(<<"inactivity">>, 10)),
+     ?_eqf(M(inactivity, infinity), T(<<"inactivity">>, <<"infinity">>)),
+     ?_eqf(M(inactivity, 10), T(<<"inactivity">>, 10)),
+     ?_eqf(M(max_wait, infinity), T(<<"max_wait">>, <<"infinity">>)),
+     ?_eqf(M(server_acks, true), T(<<"server_acks">>, true)),
+     ?_eqf(M(server_acks, false), T(<<"server_acks">>, false)),
+     ?_eqf(M(backend, mnesia), T(<<"backend">>, <<"mnesia">>)),
+     ?errf(T(<<"inactivity">>, -1)),
+     ?errf(T(<<"inactivity">>, <<"10">>)),
+     ?errf(T(<<"inactivity">>, <<"inactivity">>)),
+     ?errf(T(<<"max_wait">>, <<"10">>)),
+     ?errf(T(<<"max_wait">>, -1)),
+     ?errf(T(<<"server_acks">>, -1)),
+     ?errf(T(<<"backend">>, <<"devnull">>))].
+
+mod_caps(_Config) ->
+    run_multi(mod_caps_cases()).
+
+mod_caps_cases() ->
+    T = fun(K, V) -> #{<<"modules">> => #{<<"mod_caps">> => #{K => V}}} end,
+    M = fun(K, V) -> modopts(mod_caps, [{K, V}]) end,
+    [?_eqf(M(cache_size, 10), T(<<"cache_size">>, 10)),
+     ?_eqf(M(cache_life_time, 10), T(<<"cache_life_time">>, 10)),
+     ?_errf(T(<<"cache_size">>, -1)),
+     ?_errf(T(<<"cache_size">>, <<"infinity">>)),
```

Comment:
I think it shouldn't be erroneous case, as per this definition:
```
validate([<<"cache_size">>, <<"mod_caps">>, <<"modules">>|_], [{cache_size, V}]) ->
  validate_non_negative_integer_or_infinity(V);
```

Reply:
good catch, validator should be `validate_non_negative_integer(V)`

Reply:
👌 corrected in docs I'm currently working on